File: Windows/universal8.1/Siminov/Connect/Model/NotificationDescriptor.cs
/** 
 * [SIMINOV FRAMEWORK]
 * Copyright [2015] [Siminov Software Solution LLP|support@siminov.com]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 **/



using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Siminov.Connect.Model
{

    /// <summary>
    /// Exposes methods to GET and SET Notification Descriptor information as per define in ApplicationDescriptor.xml file by application.
    /// Example:
    ///
    ///    <siminov>
	///    
    ///        <!-- Notification Descriptor -->
    ///        <notification-descriptor>
	///        
    ///                <!-- Optional Field -->
    ///            <property name="name_of_property">value_of_property</property>
	///
    ///        </notification-descriptor>
	///    
    ///    </siminov>
    /// 
    /// </summary>
    public class NotificationDescriptor : Core.Model.IDescriptor
    {
        private IDictionary<String, String> properties = new Dictionary<String, String>();

        public IEnumerator<String> GetProperties()
        {
            return this.properties.Keys.GetEnumerator();
        }

        public String GetProperty(String name)
        {
            return this.properties[name];
        }

        public bool ContainProperty(String name)
        {
            return this.properties.ContainsKey(name);
        }

        public void AddProperty(String name, String value)
        {
            this.properties.Add(name, value);
        }

        public void RemoveProperty(String name)
        {
            this.properties.Remove(name);
        }
    }
}

File: Android/Connect/src/main/java/siminov/connect/model/NotificationDescriptor.java
/**
 * [SIMINOV FRAMEWORK - CONNECT]
 * Copyright [2014-2016] [Siminov Software Solution LLP|support@siminov.com]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 **/

package siminov.connect.model;

import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;

import siminov.core.model.IDescriptor;


/**
 * Exposes methods to GET and SET Notification Descriptor information as per define in ApplicationDescriptor.xml file by application.
 *
 *	<p>
		<pre>
		
Example:
	{@code

	<siminov>
	    
	    <!-- Notification Descriptor -->
	    <notification-descriptor>
	        
	        	<!-- Optional Field -->
	        <property name="name_of_property">value_of_property</property>
	
	    </notification-descriptor>
	    
	</siminov>
	
	}
	
		</pre>
	</p>
 */
public class NotificationDescriptor implements IDescriptor {

	private Map<String, String> properties = new HashMap<String, String> ();

	public Iterator<String> getProperties() {
		return this.properties.keySet().iterator();
	}

	public String getProperty(String name) {
		return this.properties.get(name);
	}

	public boolean containProperty(String name) {
		return this.properties.containsKey(name);
	}

	public void addProperty(String name, String value) {
		this.properties.put(name, value);
	}

	public void removeProperty(String name) {
		this.properties.remove(name);
	}
}
